frontend/packages/console-plugin-sdk/src/store.ts

```diff
@@ -171,7 +171,12 @@ export class PluginStore {
   }
 
   private updateExtensions() {
-    const dynamicPlugins = Array.from(this.loadedDynamicPlugins.values());
+    const allowedDynamicPluginNames = Array.from(this.allowedDynamicPluginNames.values());
+    const dynamicPlugins = Array.from(this.loadedDynamicPlugins.values()).sort((a, b) => {
+      const indexA = allowedDynamicPluginNames.indexOf(a.manifest.name);
+      const indexB = allowedDynamicPluginNames.indexOf(b.manifest.name);
+      return indexA - indexB;
+    });
```

Comment:
I think this is fine, but would it be simpler to just filter out plugins that aren't loaded from `allowedDynamicPluginNames` since it's already sorted?

```suggestion
    const dynamicPlugins = allowedDynamicPluginNames.filter((plugin) => {
      // return true if this.loadedDynamicPlugins contains plugin...
    });
```

Reply:
I think it would be better to keep this as is. I originally agreed that your approach would be simpler, but as I got into implementation, it got a little complicated. With the new approach, `dynamicPlugins` becomes `string[]` instead of `LoadedDynamicPlugin[]`. It would be possible to write an algorithm to transform the list of strings into a list of LoadedDynamicPlugins in the order we want, but that introduces more complexity. It's definitely possible to make this work but I think it would make this algorithm less readable.